### ISSUE TITLE
run_cimOperations.py returns different property counts on one test

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#477]
We are getting some issues between the number of properties returned in instances vs in classes in one of the run_cimoperation.py tests trying to test the deepinheritance on calls.    While I think this is an issue with the characteristics of the localonly (which is deprecated and therefore nobody knows exactly what results it produces, our only goal is really to just confirm that the deepinheritance parameter is pased to the server so we will probably just rewrite the test.

**Comment by andy-maier** [comment]
Please provide details.

**Comment by KSchopmeyer** [comment]
This is apparently an issue in Pegasus where, for whatever reason, the EnumerateInstances on CIM_ComputerSystem with  LocalOnly = True is returning properties from the superclasses:

``

WARN: cls property_count 5 != inst_property_count 9
diff set([u'Status', u'Dedicated', u'Description', u'ResetCapability', u'ElementName', u'Caption', u'OperationalStatus', u'CreationClassName', u'OtherDedicatedDescriptions', u'Name'])
ClassPropertyName=NameFormat
ClassPropertyName=Dedicated
ClassPropertyName=OtherDedicatedDescriptions
ClassPropertyName=ResetCapability
ClassPropertyName=PowerManagementCapabilities
InstancePropertyName=Caption
InstancePropertyName=Description
InstancePropertyName=ElementName
InstancePropertyName=OperationalStatus
InstancePropertyName=Status
InstancePropertyName=CreationClassName
InstancePropertyName=Name
InstancePropertyName=NameFormat
InstancePropertyName=PowerManagementCapabilities

I propose that we make this a Pegasus Issue and I will document it as a pegasus bug.  Also, we can close this issue.

**Comment by andy-maier** [comment]
Actually, with LocalOnly being deprecated for EnumerateInstance*() and GetInstance() since DSP0200 1.2, I don't think you necessarily need to consider this a Pegasus issue.

The DSP0200 1.0 definition was based upon instances (see Annex B in DSP0200 >=1.2), so please double check which interpretation of LocalOnly the current Pegasus behavior actually violates.

**Comment by KSchopmeyer** [comment]
Lets leave this one open while we sort out what pegasus is doing.

**Comment by andy-maier** [comment]
There are two action items for Karl:
* Change run_cim_operations.py test to tolerate current Pegasus behavior.
* Figure out what the desired behavior of Pegasus should be (and possibly open a Pegasus issue).

**Comment by KSchopmeyer** [comment]
For OpenPegasus the method request for getinstance with no localonly component

cimcli gi Pywbem_person -i

HTTP:

```
<IMETHODCALL NAME="GetInstance">
<LOCALNAMESPACEPATH>
<NAMESPACE NAME="root"/>
<NAMESPACE NAME="cimv2"/>
</LOCALNAMESPACEPATH>
<IPARAMVALUE NAME="InstanceName">
<INSTANCENAME CLASSNAME="PyWBEM_Person">
<KEYBINDING NAME="CreationClassName">
<KEYVALUE VALUETYPE="string">PyWBEM_Person</KEYVALUE>
</KEYBINDING>
<KEYBINDING NAME="Name">
<KEYVALUE VALUETYPE="string">Alice</KEYVALUE>
</KEYBINDING>
</INSTANCENAME>
</IPARAMVALUE>
</IMETHODCALL>
</SIMPLEREQ>
</MESSAGE>
</CIM>

```
returns

```

instance of PyWBEM_Person
{
    InstanceID = NULL;
    Caption = NULL;
    Description = NULL;
    ElementName = NULL;
    GivenName = NULL;
    Surname = NULL;
    Mail = NULL;
    UserID = NULL;
    LocalityName = NULL;
    PostalAddress = NULL;
    StateOrProvince = NULL;
    PostalCode = NULL;
    TelephoneNumber = NULL;
    BusinessCategory = NULL;
    CommonName = NULL;
    EmployeeNumber = NULL;
    EmployeeType = NULL;
    FacsimileTelephoneNumber = NULL;
    HomePhone = NULL;
    HomePostalAddress = NULL;
    JPEGPhoto = NULL;
    Manager = NULL;
    Mobile = NULL;
    OU = NULL;
    Pager = NULL;
    PreferredLanguage = NULL;
    Secretary = NULL;
    Title = NULL;
    CreationClassName = "PyWBEM_Person";
    Name = "Alice";
};

```

So the default is to operate as if LocalOnly is False

Also, it appears that the lo only option does not get passed up from the client:  When I set --lo in

cimcli gi Pywbem_person -i -lo

The server received:

```
<IMETHODCALL NAME="GetInstance">
<LOCALNAMESPACEPATH>
<NAMESPACE NAME="root"/>
<NAMESPACE NAME="cimv2"/>
</LOCALNAMESPACEPATH>
<IPARAMVALUE NAME="InstanceName">
<INSTANCENAME CLASSNAME="PyWBEM_Person">
<KEYBINDING NAME="CreationClassName">
<KEYVALUE VALUETYPE="string">PyWBEM_Person</KEYVALUE>
</KEYBINDING>
<KEYBINDING NAME="Name">
<KEYVALUE VALUETYPE="string">Alice</KEYVALUE>
</KEYBINDING>
</INSTANCENAME>
</IPARAMVALUE>
</IMETHODCALL>
</SIMPLEREQ>
</MESSAGE>
</CIM>
```

**Comment by andy-maier** [comment]
We agree that this behavior is correct w.r.t. DSP0200 and there is no OpenPegasus issue here.

The remaining issue is to make the pywbem run_cim_operations.py tolerant against the different behaviors for LocalOnly that WBEM servers can legally implement.

**Comment by KSchopmeyer** [comment]
Fixed with pr # 1369.  This one should be closed I think. Note that it was marked fixes #477 so maybe github burped.
